src/app/middleware.ts

```diff
@@ -261,3 +263,139 @@ startListening({
     );
   },
 });
+
+// Telemetry
+startListening({
+  matcher: isAnyOf(
+    chatAskQuestionThunk.rejected.match,
+    chatAskQuestionThunk.fulfilled.match,
+    diffApi.endpoints.patchSingleFileFromTicket.matchFulfilled,
+    diffApi.endpoints.patchSingleFileFromTicket.matchRejected,
+    // give files api
+    pathApi.endpoints.getFullPath.matchFulfilled,
+    pathApi.endpoints.getFullPath.matchRejected,
+    pathApi.endpoints.customizationPath.matchFulfilled,
+    pathApi.endpoints.customizationPath.matchRejected,
+    pathApi.endpoints.privacyPath.matchFulfilled,
+    pathApi.endpoints.privacyPath.matchRejected,
+    pathApi.endpoints.bringYourOwnKeyPath.matchFulfilled,
+    pathApi.endpoints.bringYourOwnKeyPath.matchRejected,
+    pathApi.endpoints.integrationsPath.matchFulfilled,
+    pathApi.endpoints.integrationsPath.matchRejected,
+  ),
+  effect: (action, listenerApi) => {
+    const state = listenerApi.getState();
+    if (chatAskQuestionThunk.rejected.match(action)) {
```

Comment:
For all of these checks, it would be more defensive to add `!action.meta.condition` because in some of the hooks, the `{skip: true}` argument will be passed and this raises a rejection but with `action.meta.condtion` set to true.